File: core/src/main/java/com/lumora/core/dto/response/ContextCompactionResponse.java
package com.lumora.core.dto.response;

import com.lumora.core.model.ContextCompaction;

public record ContextCompactionResponse(
        int beforeTokens,
        int afterTokens,
        Integer throughSequence,
        Integer retainedFromSequence,
        TokenUsageResponse usage
) {
    public static ContextCompactionResponse from(ContextCompaction value) {
        return new ContextCompactionResponse(
                value.beforeTokens(), value.afterTokens(),
                value.throughSequence(), value.retainedFromSequence(),
                new TokenUsageResponse(
                        value.usage().getPromptTokens(),
                        value.usage().getCompletionTokens(),
                        value.usage().getTotalTokens()
                )
        );
    }
}

File: agent/app/dto/response/context_compaction_response.py
from pydantic import BaseModel, ConfigDict, Field

from app.dto.response.chat_completion_response import TokenUsageResponse


class ContextCompactionResponse(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    summary: str
    before_tokens: int = Field(alias="beforeTokens")
    after_tokens: int = Field(alias="afterTokens")
    through_sequence: int | None = Field(alias="throughSequence")
    retained_from_sequence: int | None = Field(alias="retainedFromSequence")
    usage: TokenUsageResponse
